sn_cli/src/subcommands/files/mod.rs

```diff
@@ -213,32 +203,56 @@ async fn upload_files(
                 bail!("Not enough balance in wallet to pay for chunk. We have {available:?} but need {required:?} to pay for the chunk");
             }
             Err(error) => {
-                warn!(
-                    "Failed to pay for chunks. Validation steps should retry this batch: {error}"
-                );
+                error!("Failed to pay for chunks: {error}");
+                recorded_pay_errors.push(error);
                 continue;
             }
         };
 
-        // Verification will be carried out later on, if being asked to.
-        // Hence no need to carry out verification within the first attempt.
-        // Just to check there were no odd errors during upload.
-        for result in join_all(upload_chunks_in_parallel(
+        // upload paid chunks
+        for join_result in join_all(upload_chunks_in_parallel(
             file_api.clone(),
             chunks_batch.to_vec(),
-            verify_as_we_upload,
+            verify_store,
             progress_bar.clone(),
             show_holders,
         ))
         .await
         {
-            let _ = result?;
+            let upload_result = join_result?;
+            if let Err(error) = upload_result {
+                error!("Failed to upload a batch: {error}");
+                recorded_upload_errors.push(error);
```

Comment:
This `result` is in fact a `Result<Result<_>>`, and the inner one was ignored before and we would rely on verification to check that it failed. In the case we had `verify_store` as `false`, then this error would remain swallowed and the marked as successful below!